array/circular_counter.py

```diff
@@ -7,42 +9,32 @@
 For example:
 Input: consider 123456789 members sitting in a circular fashion,
 Output: 369485271
-"""
 
-a = ['1','2','3','4','5','6','7','8','9']
-
-def josepheus(int_list, skip):
-  skip = skip - 1 #list starts with 0 index
-  idx = 0
-  len_list = (len(int_list))
-  while len_list>0:
-    idx = (skip+idx)%len_list #hashing to keep changing the index to every 3rd
-    print(int_list.pop(idx))
-    len_list -= 1
 
+>>> a = ['1', '2', '3', '4', '5', '6', '7', '8', '9']
+>>> josepheus_generator = josepheus(a, 3)
+>>> assert next(josepheus_generator) == '3'
+>>> assert next(josepheus_generator) == '6'
+>>> assert next(josepheus_generator) == '9'
+>>> assert next(josepheus_generator) == '4'
+>>> assert next(josepheus_generator) == '8'
+>>> assert next(josepheus_generator) == '5'
+>>> assert next(josepheus_generator) == '2'
+>>> assert next(josepheus_generator) == '7'
+>>> assert next(josepheus_generator) == '1'
```

Comment:
`unittest` is preferred over `doctests`, as many algorithms already have unittests and it would be good to have consistency.